app/src/main/java/com/google/droidjump/GameView.java

```diff
@@ -66,23 +72,38 @@ public void run() {
             drawScene();
             sleep();
             timePoint++;
+            intervalTimePoint++;
         }
     }
 
     private void receiveLevelDetails() {
         // TODO: Serialize current level data and put it in some container
         levelTimePoints = 200;
-        levelSpeed = 50;
+        levelSpeed = levelData.getBaseSpeed();
     }
 
     public void updateGameState() {
         // TODO: Check if time point is in level data and add data to some container, than move
```

Comment:
since you have added checkTimePoint(), is there still a need of the TODO?

Reply:
Removed